### PR TITLE
feat(copy): improve autocapture and custom insight date range copy

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

The autocapture description in settings [isn't accurate](https://posthog.slack.com/archives/C02E3BKC78F/p1690448666026269) and the tooltip for [custom insight date range isn't clear](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1690394290863519).

## Changes

This PR changes that.

## How did you test this code?

No tests.

### CODE REVIEW

File: frontend/src/lib/components/DateFilter/DateFilter.tsx [hunk 1]
```diff
@@ -101,7 +101,7 @@ export function DateFilter({
                     const dateValue = dateFilterToText(
                         values[0],
                         values[1],
-                        'No date selected',
+                        'Date range specified in the respective Insight',
```
Comment (Twixes) [inline frontend/src/lib/components/DateFilter/DateFilter.tsx:104]: The tweaks around autocapture look good (I pushed a couple more though, so let me know if that makes sense), but I don't _think_ this solves the issue with the dashboard time range selector being confusing. There, the problem seems to be "Custom" being shown, and not "No date selected" (which I actually don't know how to get to show up anywhere)

Comment (thmsobrmlr) [reply frontend/src/lib/components/DateFilter/DateFilter.tsx:104]: Cool, thanks for the changes! Didn't want to change the "Custom" label here as it looks like a more complicated change. Will do this in a follow up PR.

Comment (thmsobrmlr) [reply frontend/src/lib/components/DateFilter/DateFilter.tsx:104]: https://github.com/PostHog/posthog/pull/16870

Comment (Twixes) [reply frontend/src/lib/components/DateFilter/DateFilter.tsx:104]: It definitely. I really don't know where this "No date selected"/"Date range specified in the respective Insight" was shown at all, do you have any pointers how to check this out?

Comment (thmsobrmlr) [reply frontend/src/lib/components/DateFilter/DateFilter.tsx:104]: Ah, got you. This is the tooltip for the "Custom" option in the dashboard date range selector.
Screenshot 2023-08-02 at 01 38 43

**Review by Twixes — APPROVED** [review]

File: frontend/src/lib/components/DateFilter/DateFilter.tsx (hunk 1 above)
Comment (Twixes) [reply frontend/src/lib/components/DateFilter/DateFilter.tsx]: Ohh, I see, didn't notice  it before

**Pull request merged**